src/main/java/core/basesyntax/service/OperationStrategyImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.service;
+
+import core.basesyntax.service.operations.Operation;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private Map<String, Operation> operationMap;
+
+    public OperationStrategyImpl(Map<String, Operation> operationMap) {
+        this.operationMap = operationMap;
+    }
+
+    @Override
+    public Operation getOperation(String operationSymbol) {
+        return operationMap.get(operationSymbol);

```

Comment:
Consider adding a check for null return values when the operation symbol is not found in the map. You could throw an exception or return a default operation to handle such cases gracefully.